src/basic/fs-util.c

```diff
@@ -1104,6 +1104,11 @@ int xopenat(int dir_fd, const char *path, int flags, mode_t mode) {
         assert(dir_fd >= 0 || dir_fd == AT_FDCWD);
         assert(path);
 
+        if (isempty(path)) {
+                assert(!FLAGS_SET(flags, O_CREAT|O_EXCL));
```

Comment:
This fails to build when `-DNDEBUG` is defined. IMO suppressing build failures with `assert` is not a good idea for this reason.

Will a fix for this be accepted?

Reply:
It would depend on what exactly the issue is and what the fix would be, please open an issue describing the problem in more detail.

Reply:
> please open an issue describing the problem in more detail.

Done: https://github.com/systemd/systemd/issues/29408